packages/transactions/src/transaction.ts

```diff
@@ -102,13 +110,14 @@ export class StacksTransaction {
     if (!tx.auth.spendingCondition) {
       throw new Error('Cannot set signature on transaction without spending condition');
     }
-    (tx.auth.spendingCondition as SingleSigSpendingCondition).signature =
-      createMessageSignature(parsedSig);
+    (tx.auth.spendingCondition as SingleSigSpendingCondition).signature = createMessageSignature(
```

Comment:
Rather than casting (`as SingleSigSpendingCondition`) which can potentially be type-unsafe, is there a runtime type guard that would inform typescript that this `tx.auth.spendingCondition` is already an instance of `SingleSigSpendingCondition`?

Reply:
this function is only used in a test, so I'm actually going to move it there, in addition to making the suggested change